### ISSUE TITLE
Add an option to choose which timeframe should be displayed by default

### ISSUE THREAD

**Issue by HellsCrimson** [muety/wakapi#391]
Add an option, either in the web interface (it could be an option different for each user or the admin choose it) or in the environment variable. And allow to  choose which timeframe should be displayed by default.
For exemple always display this week instead of the current day.

**Comment by muety** [comment]
Hi @HellsCrimson, thanks for your feedback. I'll add this as a low-prio feature request. A temporary solution would be to bookmark the respective URL.

**Comment by Californ1a** [comment]
I was super confused for a few minutes when I first imported my wakatime data because of this. Made me think it only imported one day's worth of stats since it showed a much lower hour count, right after clicking on dashboard, compared to the 7 days wakatime shows by default.

At the very least, I think if you change it, it should stay on that value for the session. Right now if you change it, then click on one of the projects, it stays the same timeframe which is great, but if you click Dashboard to go back rather than using the browser's back button, it reverts back to 1 day.

**Comment by muety** [comment]
Thanks for your feedback. 

I think I'd rather opt out from the proposed solution of being able to explicitly configure the default time frame. But will implement the time frame in a way that it is remembered across page changes.

**Comment by Daste745** [comment]
I can work on this. I think that we can hold the last used interval in the browser's local storage as something like `wakapi_summary_interval`. Whether to always store the interval selected by the `interval` query param, or only do that when clicked manually (through the interval picker) is your decision though. I think that always overwriting the last used time interval is a more natural approach, as the user which clicks a link (even from an external source or a bookmark) will see the same summary when they come back again.

I'll need some help here, because I don't know how to easily change the selected time interval for data fetching without updating the `window.location`. From what I can tell, it's the only option now, because changing only the return value of the `currentInterval` getter in `components/summary.js` doesn't influence the actual time range displayed on the summary page, when query params are empty.

I made a quick prototype. When there is `wakapi_summary_data` in the browser's local storage, it adds the `interval` query param to `window.location`. This *works*, but causes an additional page reload. I also dislike the mutation being in a getter function, but can't quite figure out where to put it.

Modified code in `static/assets/js/components/summary.js`:
```js
PetiteVue.createApp({
    $delimiters: ['${', '}'],
    get currentInterval() {
        const urlParams = new URLSearchParams(window.location.search)
        if (urlParams.has('interval')) {
            window.localStorage.setItem('wakapi_summary_interval', urlParams.get('interval'))
            return urlParams.get('interval')
        }

        const localStorageInterval = window.localStorage.getItem("wakapi_summary_interval")
        if (localStorageInterval) {
            window.location += `?interval=${localStorageInterval}`
            return localStorageInterval
        }

        if (!urlParams.has('from') && !urlParams.has('to')) return 'today'
        return null
    }
}).mount('#summary-page')
```

I can open a pull request with these changes, but I'm posting them here first to avoid any confusion and get feedback where to place this new logic.

**Comment by Daste745** [comment]
Here's a quick demo:
![2022-10-31_00-08-05](https://user-images.githubusercontent.com/25880357/198906349-a36864bc-9028-46ed-b54d-c3a6aee63a4f.gif)

**Comment by muety** [comment]
Hi @Daste745, thanks for looking into this! 

> [...] always overwriting the last used time interval is a more natural approach [...] 

I agree!

> [...] From what I can tell, it's the only option now [...]

You're right. Since (almost) everything in Wakapi is server-rendered, the only way to fetch data for a certain interval is to actually update the page.

One option could be: the user logs in, sees the default summary page, then JavaScript kicks in, checks localStorage for the last viewed interval and updates the URL, resulting in a page reload. This wouldn't be a nice solution, though, because of the additional redirect. The user would see the default (i.e. today) summary for a short second before actually being redirected to the correct time frame. So you can't wait for JavaScript to handle this, because of that aforementioned, ugly "content flash".

Would you would need is a mechanism that allows the server to decide which interval to present to the user, even before the page is first rendered. 

One option here could be to actually always persist the last visited interval in the database. However, I don't like that either, because (a) I think it's overkill and (b) because the database shouldn't have to mess with UI-specific data / properties. 

Long story short: **I propose to use a cookie for this** and get the best of both of the above worlds. Cookies are only stored client-side, can nevertheless be read by the server (because automatically included with every request) and don't require any JavaScript to read them. There would be a cookie, that is _set_ from JavaScript, essentially the exact same way how you're currently setting the localStorage attribute. But then, upon requesting the `/summary` route, the backend would (1.) check if a query param is explicitly set (what we already had so far) and (2.) if not, check if the `summary_interval` cookie is present.

**Comment by muety** [comment]
Check [this method](https://github.com/muety/wakapi/blob/9586dbf781c9f9664b200cd2b2246d0f388af652/routes/summary.go#L38), that'll be your starting point on the backend side. Thanks again for looking into this!
